integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,325 @@
+		assertThat(response.getContentAsString(), notNullValue());
+		
+	}
+	
+	@Test
+	public void shouldThrow404ForNonExistingTaskAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Task/" + WRONG_TASK_UUID)
+		        .accept(BaseFhirIntegrationTest.FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+	}
+	
+	@Test
+	public void shouldReturnExistingTaskAsXML() throws Exception {
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(BaseFhirIntegrationTest.FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());	
+	}
+	
+	@Test
+	public void shouldThrow404ForNonExistingTaskAsXML() throws Exception {
+		MockHttpServletResponse response = get("/Task/" + WRONG_TASK_UUID).accept(BaseFhirIntegrationTest.FhirMediaTypes.XML)
+		        .go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(BaseFhirIntegrationTest.FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+	}
+	
+	@Test
+	public void shouldCreateNewTaskAsJson() throws Exception {
+		String jsonTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_CREATE_TASK_PATH)) {
+			Objects.requireNonNull(is);
+			jsonTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		// create task
+		MockHttpServletResponse response = post("/Task").accept(FhirMediaTypes.JSON).jsonContent(jsonTask).go();
+		//lets verify the created task 
+		assertThat(response, isCreated());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+	Task task = readResponse(response);
+		assertThat(task, notNullValue());
+		
+		// try to get new task
+		response = get(task.getId()).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		Task task2 = readResponse(response);
+		
+		assertThat(task2.getId(), equalTo(task.getId()));
+		
+	}
+	
+	@Test
+	public void shouldCreateNewTaskAsXml() throws Exception {
+		
+		String xmlTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(XML_CREATE_TASK)) {
+			Objects.requireNonNull(is);
+			xmlTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		// create task
+		MockHttpServletResponse response = post("/Task").accept(FhirMediaTypes.XML).xmlContext(xmlTask).go();
+		// verify created correctly
+		
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Task task = readResponse(response);
+		assertThat(task, notNullValue());
+		assertThat(task.getStatus(), is(Task.TaskStatus.ACCEPTED));
```

Comment:
```suggestion
		assertThat(patient.getIdElement().getIdPart(), notNullValue());
		assertThat(task.getStatus(), is(Task.TaskStatus.ACCEPTED));
```

We should also verify any other fields we expect to be set on the Task.